src/Engines/ReduxEngine.js

```diff
@@ -0,0 +1,53 @@
+import { connect } from 'react-redux';
+
+class StateManager {
+  stateChangeHandler = null;
+  reducer = null;
+  reduxDispatch = null;
+
+  constructor(dispatch) {
+    this.reduxDispatch = dispatch;
```

Comment:
The Redux setup is actually more complicated then the MobX one right now for the sole reason - I need this and because of needing to provide the reducer.

In a redux app you would need to:

1. Wire up your reducer:
```
const store = createStore(combineReducer({
    nav: navReducer, // <- navReducer is imported off of ReduxEngine
    {...other reducers here}
}))
```

2. Connect your `Router` component. I think this is the part I struggle with the most. Note that I don't even pass a mapStateToProps or mapDispatchToProps
```
connect()(({ dispatch }) => <Router engine={ReduxEngine(dispatch)} {...}
```